### PR TITLE
#6 Add deep equals function with tests

### PR DESCRIPTION
**Pull request by jmeline**
There is probably a lot of work documenting this feature along with possible edge cases. Since I didn't see a contributors.md, I am not quite sure what styling expectations you may have.

Cheers!

### CODE REVIEW

File: deep.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Determines whether two objects are equal. Works on nested structures.
+
+@param {Object} obj1 can be either an object or array
+@param {Object} obj1 can be either an object or array
+
+@returns {Boolean}
+*/
+
+export {
```
Comment (jmeline) [inline deep.js:1]: Probably should be at the top of the file.

File: deep.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        if (!deepEqual(obj1[prop], obj2[prop])) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+};
+
+const isObject = obj => obj === Object(obj);
```
Comment (jmeline) [inline deep.js:51]: is there a better place for these type of utility functions?

File: deep.js (hunk 1 above)
Comment (GrosSacASac) [reply deep.js:1]: put jsdoc above the or inside the function, export at the very top

File: deep.js (hunk 2 above)
Comment (GrosSacASac) [reply deep.js:51]: it is fine

**Comment by GrosSacASac** [comment]
The purpose of this issue is to have a baseline implementation, we can always incrementally improve later.

**Comment by GrosSacASac** [comment]
Ping me when you are ready to merge

**Comment by jmeline** [comment]
I'm good for you to merge it whenever you're ready.

File: deep.js (hunk 1 above)
Comment (jmeline) [reply deep.js:1]: fixed
